src/main/java/com/example/manager/controller/UserController.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+
+    public User register(User user) {
+        return repo.save(user);
+    }
+
+    public boolean checkIfUsernameTaken(String username) {
+        return repo.existsById(username);
+    }
+
+
+    public boolean login(String username, String password) {
+        User temp = null;
+        try {
+            temp = repo.findById(username).get();
+        } catch (Exception e) {
+
+        }
+        if (temp != null) {
+            if (passwordEncoder.matches(password, temp.getPassword()) && temp.getAccountNonLocked()) {
+                resetFailedAttempts(username);
+                return true;
+            } else {
+                unlockAfterTimeExpired(username);
+                temp = repo.findById(username).get();
+                if (temp.getAccountNonLocked() && temp.getFailedAttempt() < maxFailedAttempts) {
+                    increaseFailedAttempts(username);
+                    temp = repo.findById(username).get();
+                    System.out.println("Failed attempt " + temp.getFailedAttempt() + "/4.");
+                } else {
+                    temp = repo.findById(username).get();
+                    if (temp.getFailedAttempt() == maxFailedAttempts) {
+                        lock(username);
+                        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+                        LocalDateTime  loginFailed = LocalDateTime.now().plusSeconds(60);
+                        System.out.println("User locked until " + dtf.format(loginFailed) + ".\n");
+                        return false;
+                    }
+                    unlockAfterTimeExpired(username);
+                }
+                return false;
+            }
+        }
+        System.out.println("Username does not match an existing one. \n Please enter a valid username\n");
+        return false;
+    }
+
+    public User changePassword(String username, String newPassword) {
+        User temp = repo.findById(username).get();
+        temp.setPassword(passwordEncoder.encode(newPassword));
+        repo.save(temp);
+        return repo.save(temp);
+    }
+
+    public void logout() {
+        System.out.println("You are logged out!\n");
+    }
+
+    public void increaseFailedAttempts(String username) {
+        User temp = repo.findById(username).get();
+        int failedAttempttempt = temp.getFailedAttempt() + 1;
+        temp.setFailedAttempt(failedAttempttempt);
+        repo.save(temp);
+    }
+
+    public void resetFailedAttempts(String username) {
+        User temp = repo.findById(username).get();
+        temp.setFailedAttempt(0);
+        repo.save(temp);
+    }
+
+    private void lock(String username) {
+        User temp = repo.findById(username).get();
+        temp.setAccountNonLocked(false);
+        temp.setLockTime(new Date());
+        repo.save(temp);
+    }
+
+    public boolean unlockAfterTimeExpired(String username) {
+        User temp = repo.findById(username).get();

```

Comment:
FYI: LocalDateTime is the newer API. 
With that you could do something like `LocalDateTime.now().isAfter(temp.getLockTime())`